### PR TITLE
Add use offset and length to copy read data.

### PR DESCRIPTION
**Pull request by MiyamuraMiyako**
add Read2 function to each platform.

### CODE REVIEW

File: serial_linux.go [hunk 1]
```diff
@@ -137,6 +137,17 @@ func (p *Port) Read(b []byte) (n int, err error) {
 	return p.f.Read(b)
 }
 
+func (p *Port) Read2(b []byte, idx, l int) (n int, err error) {
+	t := make([]byte, l)
+	rcnt, err := p.f.Read(t)
+
+	for i := 0; i < rcnt; i++ {
+		b[i+idx] = t[i]
```
Comment (tarm) [inline serial_linux.go]: this should be the built-in copy()

File: serial_linux.go [hunk 2]
```diff
@@ -137,6 +137,17 @@ func (p *Port) Read(b []byte) (n int, err error) {
...
+	}
+
+	return rcnt, err
```
Comment (tarm) [inline serial_linux.go]: Probably this should be `rcnt - idx`

**Comment by tarm** [comment]
Can you elaborate on why this should be part of the package?

If the caller wants to discard the first few bytes, then the caller can do that.  Add this point I don't understand why this should be part of the package.

**Comment by MiyamuraMiyako** [comment]
Easy to read multiple and sequentially into the slice.

File: serial_linux.go (hunk 1 above)
Comment (MiyamuraMiyako) [reply serial_linux.go]: You are right, I think may be do this change: ```copy(buf[idx:], tbuf[:rcnt])```

File: serial_linux.go (hunk 2 above)
Comment (MiyamuraMiyako) [reply serial_linux.go]: No, ```rcnt``` represents the length of this read.

**Comment by MiyamuraMiyako** [comment]
example:
image

I want this function because of the pcomm library in windows: [PCOMM.H Line105](https://github.com/dopplerimager/sdiexternal/blob/master/PCOMM.H#L105)

May be my solution not perfect.

**Comment by xiegeo** [comment]
If I understand the problem correctly, instead of doing `Read2(buf, start, length)`, you can do `Read(buf[start:start+length])`

**Comment by MiyamuraMiyako** [comment]
Oh, you are right.  It seems this pr can be closed

**Pull request closed without merging**